src/main/java/Commands/ParseCommands.java

```diff
@@ -0,0 +1,129 @@
+package Commands;
+
+import LuciferExceptions.InvalidException;
+import LuciferExceptions.OutOfBoundsException;
+import LuciferExceptions.WrongFormatException;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
+import TaskList.TaskList;
+import UserInterface.Ui;
+
+/**
+ * Class ParseCommands which encapsulates the command.
+ * @author Yu Meng
+ */
+public class ParseCommands {
+    /** A checker to check if it is within Lucifer chatbot commands **/
+    private static final String[] WORD_COMMANDS = {"todo", "deadline", "event", "list", "mark", "unmark", "delete", "find"};
+    /** TaskList which stores the list of user tasks. **/
+    private final TaskList task;
+
+    /**
+     * Constructor for Class ParseCommands.
+     *
+     * @param task the class that stores the list of tasks
+     */
+    public ParseCommands(TaskList task) {
+        this.task = task;
+    }
+
+    /**
+     * Parses the command from input.
+     */
+    public void parseCommand(String command) {
```

Comment:
This method seems a little lengthy and is deeply nested, making it a little difficult to understand the proper flow. Perhaps a more concise method/abstraction methods could be employed!